tests/CommunityToolkit.Aspire.Hosting.SurrealDb.Tests/SurrealDbFunctionalTests.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting;
+using Aspire.Hosting.Utils;
+using Bogus;
+using CommunityToolkit.Aspire.Testing;
+using FluentAssertions;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using SurrealDb.Net;
+using Xunit.Abstractions;
+using SurrealRecord = SurrealDb.Net.Models.Record;
+
+namespace CommunityToolkit.Aspire.Hosting.SurrealDb.Tests;
+
+[RequiresDocker]
+public class SurrealDbFunctionalTests(ITestOutputHelper testOutputHelper)
```

Comment:
I'm cautious on the maintainability of these tests, as we've had some problems in the past with this style in addition to the ones that use the examples app host project.

Once we're able to run CI on them, let's evaluate properly, but also consider what these tests are testing that the functionality the Community Toolkit provides vs what is .NET Aspire. For example, if the volume persistence fails, is that a Toolkit bug or Aspire bug? Assuming all our tests cover setting the right annotations isn't the Toolkit doing the right stuff?

Reply:
I can't really tell. But at least it can confirm that the database integration works as expected for those scenarii.